src/app/test_executive/mina_peers_test.ml

```diff
@@ -0,0 +1,66 @@
+open Core
+open Integration_test_lib
+open Currency
+
+module Make (Engine : Engine_intf) = struct
+  open Engine
+
+  (* TODO: find a way to avoid this type alias (first class module signatures restrictions make this tricky) *)
+  type network = Network.t
+
+  type log_engine = Log_engine.t
+
+  let config =
+    let open Test_config in
+    let open Test_config.Block_producer in
+    let timing : Coda_base.Account_timing.t =
+      Timed
+        { initial_minimum_balance= Balance.of_int 1000
+        ; cliff_time= Coda_numbers.Global_slot.of_int 4
+        ; vesting_period= Coda_numbers.Global_slot.of_int 2
+        ; vesting_increment= Amount.of_int 50_000_000_000 }
+    in
+    { default with
+      block_producers=
+        [ {balance= "1000"; timing}
+        ; {balance= "1000"; timing}
+        ; {balance= "1000"; timing} ]
+    ; num_snark_workers= 0 }
+
+  let run network log_engine =
+    let open Network in
+    let open Malleable_error.Let_syntax in
+    let logger = Logger.create () in
+    [%log info] "mina_peers_test: started" ;
+    let wait_for_init_partial node =
+      Log_engine.wait_for_init node log_engine
+    in
+    let%bind () =
+      Malleable_error.List.iter network.block_producers
+        ~f:wait_for_init_partial
+    in
+    [%log info] "mina_peers_test: done waiting for initialization" ;
+    let peer_list = network.block_producers in
+    (* [%log info] "peers_list"
+      ~metadata:
+      [("namespace", `String t.namespace); ("pod_id", `String t.pod_id)] ; *)
+    let get_peer_id_partial = Node.get_peer_id ~logger in
+    let%bind query_results =
+      Malleable_error.List.map peer_list ~f:get_peer_id_partial
+    in
+    [%log info] "mina_peers_test: successfully made graphql query" ;
+    (* query_results is of type (string * string sexp_list) sexp_list *)
```

Comment:
`sexp_list` is an artifact of Merlin, it's just `list`, really.

It might be better to annotate the value with the type, than to put it in a comment
That way, the validity of the type is checked by the compiler